### ISSUE TITLE
VAPOR disappears upon closing notice board

### ISSUE THREAD

**Issue by shaomeng** [NCAR/VAPOR#2825]
To reproduce:
1. Open VAPOR on macOS by right click -> open
2. Click "not check for notice" on the notice board.
3. Click "close" on the notice board.
4. VAPOR disappears. 

Note: the following screen recording illustrates this behavior. 
https://drive.google.com/file/d/1J-2l26_pdAFEQ8Qb7NTt-VjCk4yWf9eq/view?usp=sharing

**Comment by clyne** [comment]
What do you mean by "disappears"? Does it crash, does the window get hidden, etc.? I can't reproduce.

**Comment by shaomeng** [comment]
Hmm, I'm not sure if it crashes or quits normally, it just disappears, as the video shows. I'm clicking icons on MacOS so there's not terminal output to help determine its status. 

Dunno what factors in its reproducibility, but I cannot use VAPOR on my MacOS.

**Comment by clyne** [comment]
You can check the process table to see if it is still running. Works for me on MacOSX 11.4 using the 3.4.1.3ad55a6ac installer.

**Comment by shaomeng** [comment]
Good point. I don't see it in `top` results, so it must quit itself.

Here I'm using the same VAPOR installer version, and macOS 11.5. Macbook Pro 2019 16-inch.

Report status on source build:
1) If source built using apple compiler and openmp disabled, the program runs well without the reported disappearing issue.
2) if source built using standard LLVM (described [here](https://docs.google.com/document/d/1nsaarW3jaZVAJFIHsERZ7MWMlms57LbErdcLgjtAkTo/edit#heading=h.p2vqktlbuajw)), VAPOR fails to launch with the following message: 
```
shaomeng@cisl-watson VAPOR-git/build % ./bin/vapor.app/Contents/MacOS/vapor
dyld: Library not loaded: @rpath/librender.dylib
  Referenced from: /Users/shaomeng/Git/VAPOR-git/build/./bin/vapor.app/Contents/MacOS/vapor
  Reason: image not found
zsh: abort      ./bin/vapor.app/Contents/MacOS/vapor
```

**Comment by clyne** [comment]
> To reproduce:
> 
> 1. Open VAPOR on macOS by right click -> open
> 2. Click "not check for notice" on the notice board.
> 3. Click "close" on the notice board.
> 4. VAPOR disappears.
> 
> https://drive.google.com/file/d/1J-2l26_pdAFEQ8Qb7NTt-VjCk4yWf9eq/view?usp=sharing

@stasj can you reproduce this?

**Comment by StasJ** [comment]
> > To reproduce:
> > 
> > 1. Open VAPOR on macOS by right click -> open
> > 2. Click "not check for notice" on the notice board.
> > 3. Click "close" on the notice board.
> > 4. VAPOR disappears.
> > 
> > https://drive.google.com/file/d/1J-2l26_pdAFEQ8Qb7NTt-VjCk4yWf9eq/view?usp=sharing
> 
> @StasJ can you reproduce this?

I cannot reproduce this. I tested with the pre-release installer on macOS Catalina and on macOS Big Sur.

https://user-images.githubusercontent.com/2772687/127063815-eceed608-0db0-4a21-b08d-40fa300edc88.mov

**Comment by sgpearse** [comment]
I can only reproduce this on my development laptop 
* CISL-FARMINGTON (Cattaliina 10.15.7)

I can't reproduce this on the following.  The notifier has correct behavior.
* CISL-ACWORTH (Catalina, 10.15.7)
* CISL-TRIANGLE (BigSur, 11.4)
* Ubuntu20 test machine
* Ubuntu18 test machine

Unfortunately, the notifier does not show up at all on the following machines.  It was enabled in all tests.
* Windows10 test machine
* CentOS test machine
* 2 personal laptops (Windows 10)

**Comment by sgpearse** [comment]
@shaomeg will try to build with/without openMP to see if that is a related issue.

@sgpearse will make a debug build and get a trace, and try to fix

**Comment by shaomeng** [comment]
Update on non-OpenMP build on Sam's Macbook:

I'm able to build and successfully launch the program with OpenMP OFF.

However, there is no notice board showing upon opening the app. The command line has this output though:
```
shaomeng@cisl-watson VAPOR-git/build-apple % ./bin/vapor.app/Contents/MacOS/vapor 
qt.qpa.window: <QNSWindow: 0x7fc014b74c90; contentView=<QNSView: 0x7fc014b74500; QCocoaWindow(0x7fc014b743e0, window=QWidgetWindow(0x7fc014b71df0, name="QGLWidgetClassWindow"))>> has active key-value observers (KVO)! These will stop working now that the window is recreated, and will result in exceptions when the observers are removed. Break in QCocoaWindow::recreateWindowIfNeeded to debug.
```

**Comment by sgpearse** [comment]
Fixed with the changes introduced in #2839.
